ui/admin/app/components/form/field/list-wrapper/select-text/index.hbs

```diff
@@ -0,0 +1,111 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+{{#let (unique-id) (unique-id) as |keyHeaderID valueHeaderID|}}
+  <Hds::Table class='list-wrapper-field' name={{@name}}>
+    <:head as |H|>
+      <H.Tr>
+        <H.Th id={{keyHeaderID}}>{{t 'form.key.label'}}</H.Th>
+        <H.Th id={{valueHeaderID}}>{{t 'form.value.label'}}</H.Th>
+        <H.Th>
+          {{t 'titles.actions'}}
+
+        </H.Th>
+      </H.Tr>
+    </:head>
+    <:body as |B|>
+      {{#each-in this.options as |select val|}}
+        <B.Tr>
+          <B.Td>
+            <Hds::Form::Select::Base
+              @value={{select}}
+              @width={{@width}}
+              disabled={{@disabled}}
+              aria-labelledby={{keyHeaderID}}
+              {{on 'change' (fn this.selectChange select val)}}
+              as |F|
+            >
+              <F.Options>
+                {{#each-in @selectOptions as |key value|}}
+                  <option value={{key}} selected={{eq select key}}>
+                    {{value}}
+                  </option>
+                {{/each-in}}
+              </F.Options>
+            </Hds::Form::Select::Base>
+          </B.Td>
+
+          <B.Td>
+            <Hds::Form::TextInput::Field
+              @value={{val}}
+              @type='text'
+              disabled={{@disabled}}
+              aria-labelledby={{valueHeaderID}}
+              {{on 'change' (fn this.updateInput select)}}
+            />
+          </B.Td>
+
+          <B.Td>
+            {{#if this.removeOptionByKey}}
+              <Hds::Button
+                data-test-remove-button={{select}}
+                @text={{t 'actions.remove'}}
+                @color='critical'
+                @icon='trash'
+                @isIconOnly={{true}}
+                disabled={{@disabled}}
+                {{on 'click' (fn this.removeOptionByKey select)}}
+              />
+            {{/if}}
+          </B.Td>
+        </B.Tr>
+      {{/each-in}}
+
+      {{#if this.showNewRow}}
+        <B.Tr>
+          <B.Td>
+            <Hds::Form::Select::Base
+              @value={{this.newOptionKey}}
+              @width={{@width}}
+              disabled={{@disabled}}
+              aria-labelledby={{keyHeaderID}}
+              {{on 'change' (set-from-event this 'newOptionKey')}}
+              as |F|
+            >
+              <F.Options>
+                <option value='' />
```

Comment:
Shouldn't this be not selectable?

Reply:
I added the text, but if we make this disabled, the select option isn't getting selected when there's one item in the list anyway. The user won't be able to select this empty option and click `add`, since we only enable `add` when there a valid option is selected from the dropdown. I updated the PR description to show this behavior.

Reply:
Oh but i meant the user should never be able to select this option right? It's not like `credential_type` where the user might want to select it as that "removes" the choice. In this case, the user would just delete the row

Reply:
ty, done